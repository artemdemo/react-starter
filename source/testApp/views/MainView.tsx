import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGlobeAfrica } from '@fortawesome/free-solid-svg-icons';
import { JsonTest } from '../containers/JsonTest/JsonTest';

export const MainView = () => {
  return (
    <>
      <div className="mb-3">
        <FontAwesomeIcon icon={faGlobeAfrica} />
        &nbsp; Main View
      </div>
      <hr />
      <JsonTest />
    </>
  );
};
