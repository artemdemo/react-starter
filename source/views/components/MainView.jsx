import React from 'react';
import Icon from '../../components/Icon/Icon';
import TestCss from '../../components/TestCss/TestCss';

const MainView = () => {
    return (
        <React.Fragment>
            <Icon name='globe' />
            &nbsp;
            Main View
            <p>&nbsp;</p>
            <TestCss />
        </React.Fragment>
    );
};

export default MainView;
