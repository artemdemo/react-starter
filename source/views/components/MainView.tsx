import React from 'react';
import Icon from '../../components/Icon/Icon';
import TestCss from '../../containers/TestCss/TestCss';
import JsonTest from '../../containers/JsonTest/JsonTest';

const MainView = () => {
    return (
        <>
            <div className='mb-3'>
                <Icon name='globe' />
                &nbsp;
                Main View
            </div>
            <div className='mb-3'>
                <TestCss />
            </div>
            <hr />
            <JsonTest />
        </>
    );
};

export default MainView;
