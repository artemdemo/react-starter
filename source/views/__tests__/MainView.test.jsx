import React from 'react';
import renderer from 'react-test-renderer';
import MainView from '../components/MainView';

jest.mock('../../components/Icon/Icon');
jest.mock('../../containers/TestCss/TestCss');

describe('<MainView>', () => {
    it('Simple render', () => {
        const tree = renderer.create(
            <MainView />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
