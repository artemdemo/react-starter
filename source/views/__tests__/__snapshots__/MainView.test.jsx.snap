// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainView> Simple render 1`] = `
Array [
  <div
    data-mock="Icon"
  >
    {
  "name": "globe"
}
  </div>,
  "  Main View",
  <p>
     
  </p>,
  <div
    data-mock="TestCss"
  />,
]
`;
